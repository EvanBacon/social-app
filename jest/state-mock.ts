import {RootStoreModel} from './../src/state/models/root-store'
import {NavigationTabModel} from './../src/state/models/navigation'
import {SessionModel} from '../src/state/models/session'
import {NavigationModel} from '../src/state/models/navigation'
import {ShellUiModel} from '../src/state/models/shell-ui'
import {MeModel} from '../src/state/models/me'
import {OnboardModel} from '../src/state/models/onboard'
import {ProfilesViewModel} from '../src/state/models/profiles-view'
import {LinkMetasViewModel} from '../src/state/models/link-metas-view'
import {MembershipsViewModel} from '../src/state/models/memberships-view'
import {FeedModel} from '../src/state/models/feed-view'
import {NotificationsViewModel} from '../src/state/models/notifications-view'
import {LRUMap} from 'lru_map'

const mockedSessionStore = {
  serialize: jest.fn(),
  hydrate: jest.fn(),
  data: {
    service: '',
    refreshJwt: '',
    accessJwt: '',
    handle: '',
    did: '',
  },
  online: false,
  attemptingConnect: false,
  rootStore: {} as RootStoreModel,
  hasSession: true,
  clear: jest.fn(),
  setState: jest.fn(),
  setOnline: jest.fn(),
  updateAuthTokens: jest.fn(),
  connect: jest.fn(),
  describeService: jest.fn(),
  login: jest.fn(),
  createAccount: jest.fn(),
  logout: jest.fn(),

  // unknown added because of the missing private methods: _connectPromise, configureApi & _connect
} as unknown as SessionModel

const mockedNavigationTabStore = {
  serialize: jest.fn(),
  hydrate: jest.fn(),
  id: 0,
  history: [
    {
      url: '',
      ts: 0,
      title: '',
      id: 0,
    },
  ],
  index: 0,
  isNewTab: false,
  current: {
    url: '',
    ts: 0,
    title: '',
    id: 0,
  },
  canGoBack: false,
  canGoForward: false,
  backTen: [
    {
      url: '',
      title: '',
      index: 0,
      id: 0,
    },
  ],
  forwardTen: [
    {
      url: '',
      title: '',
      index: 0,
      id: 0,
    },
  ],
  navigate: jest.fn(),
  refresh: jest.fn(),
  goBack: jest.fn(),
  fixedTabReset: jest.fn(),
  goForward: jest.fn(),
  goToIndex: jest.fn(),
  setTitle: jest.fn(),
  setIsNewTab: jest.fn(),
  fixedTabPurpose: 0,
  getBackList: jest.fn(),
  getForwardList: jest.fn(),
} as NavigationTabModel

const mockedNavigationStore = {
  serialize: jest.fn(),
  hydrate: jest.fn(),
  tabs: [],
  tabIndex: 0,
  clear: jest.fn(),
  tab: mockedNavigationTabStore,
  tabCount: 1,
  isCurrentScreen: jest.fn(),
  navigate: jest.fn(),
  refresh: jest.fn(),
  setTitle: jest.fn(),
  handleLink: jest.fn(),
  switchTo: jest.fn(),
  setActiveTab: jest.fn(),
  closeTab: jest.fn(),
  newTab: jest.fn(),
} as NavigationModel

const mockedShellStore = {
  minimalShellMode: false,
  isMainMenuOpen: false,
  isModalActive: false,
  activeModal: undefined,
  isLightboxActive: false,
  activeLightbox: undefined,
  isComposerActive: false,
  composerOpts: undefined,
  setMainMenuOpen: jest.fn(),
  setMinimalShellMode: jest.fn(),
  openModal: jest.fn(),
  closeModal: jest.fn(),
  closeComposer: jest.fn(),
  closeLightbox: jest.fn(),
  openComposer: jest.fn(),
  openLightbox: jest.fn(),
} as ShellUiModel

const mockedMeStore = {
  serialize: jest.fn(),
  hydrate: jest.fn(),
  did: '',
  handle: '',
  displayName: '',
  description: '',
  avatar: '',
  notificationCount: 0,
  rootStore: {} as RootStoreModel,
  // TODO: mock these models
  memberships: {
    placeholder: null,
  } as MembershipsViewModel,
  mainFeed: {
    placeholder: null,
  } as FeedModel,
  notifications: {
    placeholder: null,
  } as NotificationsViewModel,
  clear: jest.fn(),
  load: jest.fn(),
  clearNotificationCount: jest.fn(),
  fetchStateUpdate: jest.fn(),
  refreshMemberships: jest.fn(),
} as MeModel

const mockedOnboardStore = {
  serialize: jest.fn(),
  hydrate: jest.fn(),
  isOnboarding: false,
  stage: '',
  start: jest.fn(),
  stop: jest.fn(),
  next: jest.fn(),
} as OnboardModel

const mockedProfileStore = {
  hydrate: jest.fn(),
  serialize: jest.fn(),
  cache: new LRUMap(100),
  rootStore: {} as RootStoreModel,
  getProfile: jest.fn(),
  overwrite: jest.fn(),
} as ProfilesViewModel

const mockedLinkMetasStore = {
  hydrate: jest.fn(),
  serialize: jest.fn(),
  cache: new LRUMap(100),
  rootStore: {} as RootStoreModel,
  getLinkMeta: jest.fn(),
} as LinkMetasViewModel

export const mockedRootStore = {
  api: false,
  resolveName: jest.fn(),
  serialize: jest.fn(),
  hydrate: jest.fn(),
  fetchStateUpdate: jest.fn(),
  clearAll: jest.fn(),
  session: mockedSessionStore,
  nav: mockedNavigationStore,
  shell: mockedShellStore,
  me: mockedMeStore,
  onboard: mockedOnboardStore,
  profiles: mockedProfileStore,
  linkMetas: mockedLinkMetasStore,
}
