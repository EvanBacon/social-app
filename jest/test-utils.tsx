import React from 'react'
import {render} from '@testing-library/react-native'
import {GestureHandlerRootView} from 'react-native-gesture-handler'
import {RootSiblingParent} from 'react-native-root-siblings'
import {SafeAreaProvider} from 'react-native-safe-area-context'
import {RootStoreModel, RootStoreProvider} from '../src/state'
import {mockedRootStore} from './state-mock'

const WrappedComponent = ({children}: any) => {
  return (
    // eslint-disable-next-line react-native/no-inline-styles
    <GestureHandlerRootView style={{flex: 1}}>
      <RootSiblingParent>
        <RootStoreProvider value={mockedRootStore as RootStoreModel}>
          <SafeAreaProvider>{children}</SafeAreaProvider>
        </RootStoreProvider>
      </RootSiblingParent>
    </GestureHandlerRootView>
  )
}

const customRender = (ui: any, options?: any) =>
  render(ui, {wrapper: WrappedComponent, ...options})

// re-export everything
export * from '@testing-library/react-native'

// override render method
export {customRender as render}
