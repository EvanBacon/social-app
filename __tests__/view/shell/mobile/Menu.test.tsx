import React from 'react'
import {Menu} from '../../../../src/view/shell/mobile/Menu'
import {render} from '../../../../jest/test-utils'

it('Menu renders correctly', () => {
  const mockedProps = {
    visible: true,
    onClose: jest.fn(),
  }
  const tree = render(<Menu {...mockedProps} />)
  expect(tree).toMatchSnapshot()
})
