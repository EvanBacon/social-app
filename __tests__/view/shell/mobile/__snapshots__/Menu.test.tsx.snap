// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu matches snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "backgroundColor": "#ffffff",
        },
      ]
    }
    testID="menuView"
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "margin": 10,
          "marginBottom": 6,
          "opacity": 1,
        }
      }
      testID="profileCardButton"
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight={60}
        bbWidth={60}
        focusable={false}
        height={60}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "flex": 0,
              "height": 60,
              "width": 60,
            },
          ]
        }
        vbHeight={100}
        vbWidth={100}
        width={60}
      >
        <RNSVGGroup>
          <RNSVGDefs>
            <RNSVGLinearGradient
              gradient={
                Array [
                  0,
                  -6862849,
                  1,
                  -16742913,
                ]
              }
              gradientTransform={null}
              gradientUnits={0}
              name="grad"
              x1="0"
              x2="1"
              y1="0"
              y2="1"
            />
          </RNSVGDefs>
          <RNSVGCircle
            cx="50"
            cy="50"
            fill={
              Array [
                1,
                "grad",
              ]
            }
            propList={
              Array [
                "fill",
              ]
            }
            r="50"
          />
          <RNSVGText
            content={null}
            dx={Array []}
            dy={Array []}
            fill={4294967295}
            font={
              Object {
                "fontSize": "50",
                "fontWeight": "bold",
                "textAnchor": "middle",
              }
            }
            propList={
              Array [
                "fill",
              ]
            }
            rotate={Array []}
            x={
              Array [
                "50",
              ]
            }
            y={
              Array [
                "67",
              ]
            }
          >
            <RNSVGTSpan
              content="T"
              dx={Array []}
              dy={Array []}
              font={Object {}}
              rotate={Array []}
              x={Array []}
              y={Array []}
            />
          </RNSVGText>
        </RNSVGGroup>
      </RNSVGSvgView>
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000000",
              },
              Object {
                "fontSize": 24,
                "fontWeight": "500",
              },
              Array [
                Object {
                  "color": "#000000",
                },
                Object {
                  "marginLeft": 12,
                },
              ],
            ]
          }
        >
          test
        </Text>
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000000",
              },
              Object {
                "fontSize": 16,
                "letterSpacing": 0.25,
              },
              Array [
                Object {
                  "color": "#645454",
                },
                Object {
                  "marginLeft": 12,
                },
              ],
            ]
          }
        >
          @
          test
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#f8f3f3",
          "borderRadius": 8,
          "flexDirection": "row",
          "margin": 10,
          "marginBottom": 0,
          "opacity": 1,
          "paddingHorizontal": 12,
          "paddingVertical": 10,
        }
      }
      testID="searchBtn"
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight={25}
        bbWidth={25}
        color={4278190080}
        fill="none"
        focusable={false}
        height={25}
        meetOrSlice={0}
        minX={0}
        minY={0}
        stroke="currentColor"
        strokeWidth={2}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "color": "#000000",
            },
            Object {
              "flex": 0,
              "height": 25,
              "width": 25,
            },
          ]
        }
        tintColor={4278190080}
        vbHeight={24}
        vbWidth={24}
        width={25}
      >
        <RNSVGGroup
          fill={null}
          propList={
            Array [
              "fill",
              "stroke",
              "strokeWidth",
            ]
          }
          stroke={
            Array [
              2,
            ]
          }
          strokeWidth={2}
        >
          <RNSVGPath
            d="M21 21l-5.197-5.197m0 0A7.5 7.5 0 105.196 5.196a7.5 7.5 0 0010.607 10.607z"
            propList={
              Array [
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            strokeLinecap={1}
            strokeLinejoin={1}
          />
        </RNSVGGroup>
      </RNSVGSvgView>
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
            },
            Object {
              "fontSize": 20,
              "fontWeight": "500",
              "letterSpacing": 0.15,
            },
            Array [
              Object {
                "color": "#000000",
              },
              Object {
                "fontWeight": "normal",
                "marginLeft": 14,
              },
            ],
          ]
        }
      >
        Search
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "borderBottomWidth": 1,
            "paddingBottom": 10,
            "paddingHorizontal": 10,
            "paddingTop": 10,
          },
          Object {
            "borderColor": "#e4e2e2",
          },
        ]
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "opacity": 1,
            "paddingLeft": 6,
            "paddingRight": 10,
            "paddingVertical": 8,
          }
        }
        testID="menuItemButton"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "height": 36,
                "justifyContent": "center",
                "marginRight": 12,
                "width": 36,
              },
            ]
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight="26"
            bbWidth="26"
            color={4278190080}
            focusable={false}
            height="26"
            meetOrSlice={0}
            minX={0}
            minY={0}
            stroke="currentColor"
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "color": "#000000",
                },
                Object {
                  "flex": 0,
                  "height": 26,
                  "width": 26,
                },
              ]
            }
            tintColor={4278190080}
            vbHeight={48}
            vbWidth={48}
            width="26"
          >
            <RNSVGGroup
              propList={
                Array [
                  "stroke",
                ]
              }
              stroke={
                Array [
                  2,
                ]
              }
            >
              <RNSVGPath
                d="M 23.951 2 C 23.631 2.011 23.323 2.124 23.072 2.322 L 8.859 13.52 C 7.055 14.941 6 17.114 6 19.41 L 6 38.5 C 6 39.864 7.136 41 8.5 41 L 18.5 41 C 19.864 41 21 39.864 21 38.5 L 21 28.5 C 21 28.205 21.205 28 21.5 28 L 26.5 28 C 26.795 28 27 28.205 27 28.5 L 27 38.5 C 27 39.864 28.136 41 29.5 41 L 39.5 41 C 40.864 41 42 39.864 42 38.5 L 42 19.41 C 42 17.114 40.945 14.941 39.141 13.52 L 24.928 2.322 C 24.65 2.103 24.304 1.989 23.951 2 Z"
                propList={
                  Array [
                    "strokeWidth",
                  ]
                }
                strokeWidth={4}
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000000",
              },
              Object {
                "fontSize": 20,
                "fontWeight": "500",
                "letterSpacing": 0.15,
              },
              Array [
                Object {
                  "color": "#000000",
                },
                Object {
                  "flex": 1,
                  "fontWeight": "normal",
                },
              ],
            ]
          }
        >
          Home
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "opacity": 1,
            "paddingLeft": 6,
            "paddingRight": 10,
            "paddingVertical": 8,
          }
        }
        testID="menuItemButton"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "height": 36,
                "justifyContent": "center",
                "marginRight": 12,
                "width": 36,
              },
            ]
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight="28"
            bbWidth="28"
            color={4278190080}
            fill="none"
            focusable={false}
            height="28"
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "color": "#000000",
                },
                Object {
                  "flex": 0,
                  "height": 28,
                  "width": 28,
                },
              ]
            }
            tintColor={4278190080}
            vbHeight={24}
            vbWidth={24}
            width="28"
          >
            <RNSVGGroup
              fill={null}
              propList={
                Array [
                  "fill",
                ]
              }
            >
              <RNSVGPath
                d="M0 0h24v24H0z"
                fill={null}
                propList={
                  Array [
                    "fill",
                  ]
                }
              />
              <RNSVGPath
                d="M20 17h2v2H2v-2h2v-7a8 8 0 1 1 16 0v7zm-2 0v-7a6 6 0 1 0-12 0v7h12zm-9 4h6v2H9v-2z"
                fill={
                  Array [
                    2,
                  ]
                }
                propList={
                  Array [
                    "fill",
                  ]
                }
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000000",
              },
              Object {
                "fontSize": 20,
                "fontWeight": "500",
                "letterSpacing": 0.15,
              },
              Array [
                Object {
                  "color": "#000000",
                },
                Object {
                  "flex": 1,
                  "fontWeight": "normal",
                },
              ],
            ]
          }
        >
          Notifications
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "borderBottomWidth": 1,
            "paddingBottom": 10,
            "paddingHorizontal": 10,
            "paddingTop": 10,
          },
          Object {
            "borderColor": "#e4e2e2",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
            },
            Object {
              "fontSize": 17,
              "fontWeight": "bold",
              "letterSpacing": 0.15,
            },
            Array [
              Object {
                "color": "#000000",
              },
              Object {
                "paddingHorizontal": 4,
                "paddingVertical": 8,
              },
            ],
          ]
        }
      >
        Scenes
      </Text>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "opacity": 1,
            "paddingLeft": 6,
            "paddingRight": 10,
            "paddingVertical": 8,
          }
        }
        testID="menuItemButton"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "height": 36,
                "justifyContent": "center",
                "marginRight": 12,
                "width": 36,
              },
            ]
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={34}
            bbWidth={34}
            focusable={false}
            height={34}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "flex": 0,
                  "height": 34,
                  "width": 34,
                },
              ]
            }
            vbHeight={100}
            vbWidth={100}
            width={34}
          >
            <RNSVGGroup>
              <RNSVGDefs>
                <RNSVGLinearGradient
                  gradient={
                    Array [
                      0,
                      -1292135,
                      1,
                      -2424577,
                    ]
                  }
                  gradientTransform={null}
                  gradientUnits={0}
                  name="grad"
                  x1="0"
                  x2="1"
                  y1="0"
                  y2="1"
                />
              </RNSVGDefs>
              <RNSVGCircle
                cx="50"
                cy="50"
                fill={
                  Array [
                    1,
                    "grad",
                  ]
                }
                propList={
                  Array [
                    "fill",
                  ]
                }
                r="50"
              />
              <RNSVGText
                content={null}
                dx={Array []}
                dy={Array []}
                fill={4294967295}
                font={
                  Object {
                    "fontSize": "50",
                    "fontWeight": "bold",
                    "textAnchor": "middle",
                  }
                }
                propList={
                  Array [
                    "fill",
                  ]
                }
                rotate={Array []}
                x={
                  Array [
                    "50",
                  ]
                }
                y={
                  Array [
                    "67",
                  ]
                }
              >
                <RNSVGTSpan
                  content="X"
                  dx={Array []}
                  dy={Array []}
                  font={Object {}}
                  rotate={Array []}
                  x={Array []}
                  y={Array []}
                />
              </RNSVGText>
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000000",
              },
              Object {
                "fontSize": 20,
                "fontWeight": "500",
                "letterSpacing": 0.15,
              },
              Array [
                Object {
                  "color": "#000000",
                },
                Object {
                  "flex": 1,
                  "fontWeight": "normal",
                },
              ],
            ]
          }
        >
          ,
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "borderBottomWidth": 1,
            "paddingBottom": 10,
            "paddingHorizontal": 10,
            "paddingTop": 10,
          },
          Object {
            "borderColor": "#e4e2e2",
          },
        ]
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "opacity": 1,
            "paddingLeft": 6,
            "paddingRight": 10,
            "paddingVertical": 8,
          }
        }
        testID="menuItemButton"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "height": 36,
                "justifyContent": "center",
                "marginRight": 12,
                "width": 36,
              },
            ]
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight="30"
            bbWidth="30"
            color={4278190080}
            fill="none"
            focusable={false}
            height="30"
            meetOrSlice={0}
            minX={0}
            minY={0}
            stroke="currentColor"
            strokeWidth={2}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "color": "#000000",
                },
                Object {
                  "flex": 0,
                  "height": 30,
                  "width": 30,
                },
              ]
            }
            tintColor={4278190080}
            vbHeight={24}
            vbWidth={24}
            width="30"
          >
            <RNSVGGroup
              fill={null}
              propList={
                Array [
                  "fill",
                  "stroke",
                  "strokeWidth",
                ]
              }
              stroke={
                Array [
                  2,
                ]
              }
              strokeWidth={2}
            >
              <RNSVGPath
                d="M18 18.72a9.094 9.094 0 003.741-.479 3 3 0 00-4.682-2.72m.94 3.198l.001.031c0 .225-.012.447-.037.666A11.944 11.944 0 0112 21c-2.17 0-4.207-.576-5.963-1.584A6.062 6.062 0 016 18.719m12 0a5.971 5.971 0 00-.941-3.197m0 0A5.995 5.995 0 0012 12.75a5.995 5.995 0 00-5.058 2.772m0 0a3 3 0 00-4.681 2.72 8.986 8.986 0 003.74.477m.94-3.197a5.971 5.971 0 00-.94 3.197M15 6.75a3 3 0 11-6 0 3 3 0 016 0zm6 3a2.25 2.25 0 11-4.5 0 2.25 2.25 0 014.5 0zm-13.5 0a2.25 2.25 0 11-4.5 0 2.25 2.25 0 014.5 0z"
                propList={
                  Array [
                    "strokeLinecap",
                    "strokeLinejoin",
                  ]
                }
                strokeLinecap={1}
                strokeLinejoin={1}
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000000",
              },
              Object {
                "fontSize": 20,
                "fontWeight": "500",
                "letterSpacing": 0.15,
              },
              Array [
                Object {
                  "color": "#000000",
                },
                Object {
                  "flex": 1,
                  "fontWeight": "normal",
                },
              ],
            ]
          }
        >
          Create a scene
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "opacity": 1,
            "paddingLeft": 6,
            "paddingRight": 10,
            "paddingVertical": 8,
          }
        }
        testID="menuItemButton"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "height": 36,
                "justifyContent": "center",
                "marginRight": 12,
                "width": 36,
              },
            ]
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight="30"
            bbWidth="30"
            color={4278190080}
            fill="none"
            focusable={false}
            height="30"
            meetOrSlice={0}
            minX={0}
            minY={0}
            stroke="currentColor"
            strokeWidth={2}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "color": "#000000",
                },
                Object {
                  "flex": 0,
                  "height": 30,
                  "width": 30,
                },
              ]
            }
            tintColor={4278190080}
            vbHeight={24}
            vbWidth={24}
            width="30"
          >
            <RNSVGGroup
              fill={null}
              propList={
                Array [
                  "fill",
                  "stroke",
                  "strokeWidth",
                ]
              }
              stroke={
                Array [
                  2,
                ]
              }
              strokeWidth={2}
            >
              <RNSVGPath
                d="M9.594 3.94c.09-.542.56-.94 1.11-.94h2.593c.55 0 1.02.398 1.11.94l.213 1.281c.063.374.313.686.645.87.074.04.147.083.22.127.324.196.72.257 1.075.124l1.217-.456a1.125 1.125 0 011.37.49l1.296 2.247a1.125 1.125 0 01-.26 1.431l-1.003.827c-.293.24-.438.613-.431.992a6.759 6.759 0 010 .255c-.007.378.138.75.43.99l1.005.828c.424.35.534.954.26 1.43l-1.298 2.247a1.125 1.125 0 01-1.369.491l-1.217-.456c-.355-.133-.75-.072-1.076.124a6.57 6.57 0 01-.22.128c-.331.183-.581.495-.644.869l-.213 1.28c-.09.543-.56.941-1.11.941h-2.594c-.55 0-1.02-.398-1.11-.94l-.213-1.281c-.062-.374-.312-.686-.644-.87a6.52 6.52 0 01-.22-.127c-.325-.196-.72-.257-1.076-.124l-1.217.456a1.125 1.125 0 01-1.369-.49l-1.297-2.247a1.125 1.125 0 01.26-1.431l1.004-.827c.292-.24.437-.613.43-.992a6.932 6.932 0 010-.255c.007-.378-.138-.75-.43-.99l-1.004-.828a1.125 1.125 0 01-.26-1.43l1.297-2.247a1.125 1.125 0 011.37-.491l1.216.456c.356.133.751.072 1.076-.124.072-.044.146-.087.22-.128.332-.183.582-.495.644-.869l.214-1.281z"
                propList={
                  Array [
                    "strokeLinecap",
                    "strokeLinejoin",
                  ]
                }
                strokeLinecap={1}
                strokeLinejoin={1}
              />
              <RNSVGPath
                d="M15 12a3 3 0 11-6 0 3 3 0 016 0z"
                propList={
                  Array [
                    "strokeLinecap",
                    "strokeLinejoin",
                  ]
                }
                strokeLinecap={1}
                strokeLinejoin={1}
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000000",
              },
              Object {
                "fontSize": 20,
                "fontWeight": "500",
                "letterSpacing": 0.15,
              },
              Array [
                Object {
                  "color": "#000000",
                },
                Object {
                  "flex": 1,
                  "fontWeight": "normal",
                },
              ],
            ]
          }
        >
          Settings
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "borderBottomWidth": 1,
            "paddingBottom": 10,
            "paddingHorizontal": 10,
            "paddingTop": 10,
          },
          Object {
            "borderColor": "#e4e2e2",
          },
        ]
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#ffffff",
            "opacity": 1,
            "paddingHorizontal": 10,
            "paddingVertical": 8,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "borderColor": "#e4e2e2",
                },
                Object {
                  "borderRadius": 15,
                  "borderWidth": 1,
                  "height": 26,
                  "marginRight": 10,
                  "padding": 4,
                  "width": 42,
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#c1b9b9",
                  },
                  Object {
                    "borderRadius": 10,
                    "height": 16,
                    "width": 16,
                  },
                  undefined,
                ]
              }
            />
          </View>
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                },
                Object {
                  "fontSize": 14,
                  "fontWeight": "500",
                  "letterSpacing": 0.5,
                },
                Array [
                  Object {
                    "color": "#000000",
                    "fontWeight": undefined,
                  },
                  Object {
                    "flex": 1,
                  },
                ],
              ]
            }
          >
            Dark mode
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "paddingHorizontal": 14,
          "paddingVertical": 18,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
            },
            Object {
              "fontSize": 16,
              "letterSpacing": 0.25,
            },
            Array [
              Object {
                "color": "#645454",
              },
            ],
          ]
        }
      >
        Build version 
         (
        )
      </Text>
    </View>
  </View>
</View>
`;
